### PR TITLE
CIRC-2109 Pass additional includeRoutingServicePoints parameter when needed

### PR DESCRIPTION
**Pull request by alexanderkurash**
[CIRC-2109](https://folio-org.atlassian.net/browse/CIRC-2109)

## Purpose
After [MODINVSTOR-1219](https://folio-org.atlassian.net/browse/MODINVSTOR-1219) was implemented, we need to pass `includeRoutingServicePoints=true` parameter to `/service-points` endpoint if we don't want routing service points to be ignored.

## Approach
A new class `CustomParamCollectionResourceClient` is allowing to add custom query parameters to the regular "collection resource" calls.

### CODE REVIEW

File: src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    return client.get(collectionRoot, pageLimit, customQueryParameter);
+  }
+
+  public CompletableFuture<Result<Response>> get(String id) {
+    return client.get(individualRecordUrl(id), customQueryParameter);
+  }
+
+  @Override
+  public CompletableFuture<Result<Response>> getMany(CqlQuery cqlQuery,
+      PageLimit pageLimit) {
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java:42]:
```suggestion
  public CompletableFuture<Result<Response>> getMany(CqlQuery cqlQuery,
    PageLimit pageLimit) {
```

File: src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    return getMany(cqlQuery, pageLimit, noOffset());
+  }
+
+  @Override
+  public CompletableFuture<Result<Response>> getMany(CqlQuery cqlQuery,
+      PageLimit pageLimit, Offset offset) {
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java:49]:
```suggestion
  public CompletableFuture<Result<Response>> getMany(CqlQuery cqlQuery,
    PageLimit pageLimit, Offset offset) {
```

File: src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java (hunk 1 above)
Comment (alexanderkurash) [reply src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java]: done

File: src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java (hunk 2 above)
Comment (alexanderkurash) [reply src/main/java/org/folio/circulation/support/CustomParamCollectionResourceClient.java]: done

**Review by OleksandrVidinieiev — APPROVED** [review]

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**